python/lsst/pipe/base/pipeline.py

```diff
@@ -83,6 +83,7 @@ class TaskDef:
     """
     def __init__(self, taskName, config, taskClass=None, label=""):
         self.taskName = taskName
+        config.freeze()
```

Comment:
I'm still not comfortable that this will have a side effect on the reference held in `toExapandedPipeline` (or any other place this may end up used). That said it is not unreasonable to ensure the config is frozen, and you did put a note in the docs, so I have no strong opinion.